### PR TITLE
(BUGZ-955) Disable unnecessary picking

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.atlassian.net/browse/BUGZ-955

This PR disables picking for certain picks in situations when they are not needed, improving performance.

### CODE REVIEW

File: scripts/system/controllers/grab.js [hunk 1]
```diff
@@ -22,6 +22,12 @@
 
 (function() { // BEGIN LOCAL_SCOPE
 
+// If we are in simplified UI, we don't need desktop grabbing.
+// So, disable for now to improve performance.
+if (Settings.getValue("simplifiedUI/keepExistingUIAndScripts") === false) {
```
Comment (zfox23) [inline scripts/system/controllers/grab.js]:
```suggestion
if (Settings.getValue("simplifiedUI/keepExistingUIAndScripts", false) === false) {
```

File: scripts/system/controllers/controllerModules/teleport.js [hunk 2]
```diff
@@ -366,7 +366,7 @@ Script.include("/~/system/libraries/controllers.js");
             var offset = _this.pickHeightOffset * capsuleRatio;
 
             _this.teleportHandCollisionPick = Picks.createPick(PickType.Collision, {
-                enabled: true,
+                enabled: false,
```
Comment (zfox23) [inline scripts/system/controllers/controllerModules/teleport.js]: how does this get re-enabled?

File: scripts/system/controllers/controllerModules/teleport.js [hunk 3]
```diff
@@ -382,7 +382,7 @@ Script.include("/~/system/libraries/controllers.js");
             });
 
             _this.teleportHeadCollisionPick = Picks.createPick(PickType.Collision, {
-                enabled: true,
+                enabled: false,
```
Comment (zfox23) [inline scripts/system/controllers/controllerModules/teleport.js]: how does this get re-enabled?

File: scripts/simplifiedUIBootstrapper.js [hunk 4]
```diff
@@ -14,6 +14,8 @@
 
 var currentlyRunningScripts = ScriptDiscoveryService.getRunning();
 
+// Make sure this setting is defined early so grab.js can decide whether it should run
+Settings.getValue("simplifiedUI/keepExistingUIAndScripts", false);
```
Comment (zfox23) [inline scripts/simplifiedUIBootstrapper.js]: What is this doing?

File: scripts/system/controllers/controllerDispatcher.js [hunk 5]
```diff
@@ -285,6 +285,21 @@ Script.include("/~/system/libraries/controllerDispatcherUtils.js");
                 }
             }
 
+            // Enable/disable controller raypicking depending on whether we are in HMD
+            if (HMD.active) {
```
Comment (zfox23) [inline scripts/system/controllers/controllerDispatcher.js]: What happens if `HMD.active` changes? Don't you need to enable/disable these picks when that value changes?

Comment (sabrina-shanman) [reply scripts/system/controllers/controllerDispatcher.js]: This is running inside the update loop.

File: scripts/simplifiedUIBootstrapper.js (hunk 4 above)
Comment (sabrina-shanman) [reply scripts/simplifiedUIBootstrapper.js]: grab.js runs before controllerDispatcher.js, which sets this value. Although now that I think about it, grab.js is self-contained, so I could just change the order of the script loading. and get rid of this.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20432/HighFidelity-Beta-PR15949-e4e2b55.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20432/HighFidelity-Beta-Interface-PR15949-e4e2b55.dmg

File: scripts/system/controllers/controllerModules/teleport.js (hunk 3 above)
Comment (sabrina-shanman) [reply scripts/system/controllers/controllerModules/teleport.js]: The run function gets called when the plugin is active, and in that function the collision picks are enabled/disabled as needed.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20432/HighFidelity-Beta-PR15949-e4e2b55.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20432/HighFidelity-Quest-Beta-PR15949-e4e2b55.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20432/HighFidelity-Beta-PR15949-e4e2b55.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20432/HighFidelity-Beta-Interface-PR15949-e4e2b55.exe

File: scripts/simplifiedUIBootstrapper.js (hunk 4 above)
Comment (zfox23) [reply scripts/simplifiedUIBootstrapper.js]: This line of code is just getting the value and not assigning that value to any variable - I still don't understand what its purpose is.

Comment (sabrina-shanman) [reply scripts/simplifiedUIBootstrapper.js]: You're right, my misunderstanding. Next implementation should fix it.

File: scripts/simplifiedUIBootstrapper.js [hunk 6]
```diff
@@ -14,6 +14,10 @@
 
 var currentlyRunningScripts = ScriptDiscoveryService.getRunning();
 
+// Make sure this setting is defined early so grab.js can decide whether it should run, regardless of the order in which the default scripts are loaded
+if (Settings.getValue("simplifiedUI/keepExistingUIAndScripts") === "") {
```
Comment (zfox23) [inline scripts/simplifiedUIBootstrapper.js]: Ah, I understand what you were going for here. Please don't do this this way. You could accomplish this same thing safely by removing this code and adding the suggestion I made here: https://github.com/highfidelity/hifi/pull/15949/files#r305102384

File: scripts/system/controllers/grab.js [hunk 7]
```diff
@@ -22,6 +22,12 @@
 
 (function() { // BEGIN LOCAL_SCOPE
 
+// If we are in simplified UI, we don't need desktop grabbing.
+// So, disable for now to improve performance.
+if (Settings.getValue("simplifiedUI/keepExistingUIAndScripts", true) === false) {
```
Comment (zfox23) [inline scripts/system/controllers/grab.js]: Please set the default value to `false`, not `true`.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20439/HighFidelity-Beta-PR15949-fc1e8fa.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20439/HighFidelity-Beta-Interface-PR15949-fc1e8fa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20439/HighFidelity-Beta-PR15949-fc1e8fa.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20439/HighFidelity-Beta-Interface-PR15949-fc1e8fa.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20439/HighFidelity-Beta-PR15949-fc1e8fa.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20439/HighFidelity-Quest-Beta-PR15949-fc1e8fa.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20455/HighFidelity-Beta-PR15949-dde0d0b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20455/HighFidelity-Beta-Interface-PR15949-dde0d0b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20455/HighFidelity-Beta-PR15949-dde0d0b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20455/HighFidelity-Quest-Beta-PR15949-dde0d0b.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20455/HighFidelity-Beta-PR15949-dde0d0b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20455/HighFidelity-Beta-Interface-PR15949-dde0d0b.exe

**Comment by sabrina-shanman** [comment]
Closing in favor of https://github.com/highfidelity/hifi/pull/15967, until there can be a better solution for disabling desktop grab.

**Pull request closed without merging**